src/pages/Admin/Venue/Rooms/RoomsForm.tsx

```diff
@@ -313,6 +314,22 @@ const RoomInnerForm: React.FC<RoomInnerForm> = (props) => {
                       <span className="slider round"></span>
                     </label>
                   </div>
+                  <div className="toggle-room">
+                    <div className="input-title">Room Type</div>
+                    <Form.Control
```

Comment:
Instead of copy/paste duplicating the code between here and `src/pages/Admin/Room/Edit/RoomEditModal.tsx`, it would be better to extract it into a new 'mini component', and then use that in both places.

Reply:
Do we plan to keep Admin as well as Admin_v2?

Reply:
Long term, no; but right now as we are transitioning and they don’t both support everything, no.

That said, duplication hurts us in the interim, and extracting sub-components is a better pattern than ending up with the 100’s+ line files our admin code currently has.

Reply:
Sounds good. I'll refactor it.

Reply:
@0xdevalias I am not sure where do I put the refactored component as I do not see it being reused anywhere once the legacy admin code is removed. Should this go in atoms or in as a component in `pages/Admin/Room`.

Reply:
@denisdimitrov @mike-lvov Do you have thoughts/opinions on @TroySK's question in https://github.com/sparkletown/sparkle/pull/1305#discussion_r612637836 ?

Reply:
I think it would be fine to put the component in the atoms or molecules directory. IMO we should do this to all of the other components that are in pages/Admin but in another PR.

Reply:
> I think it would be fine to put the component in the atoms or molecules directory. IMO we should do this to all of the other components that are in pages/Admin but in another PR.

I would still expect that there are cases when pages/venues have their own components, which are not shared because they are quite specific. For example `ChatSidebar` has `PrivateChatPreview` component, which I don't think can be reused anywhere else. And if in the future there is a case, where we would actually want to reuse it, it's quite easy to just drag that component into `shared` components.

Thus, I would say that we should not just `move all of the components` into `atoms/molecules`.

@denisdimitrov do you think all of those components will be reused anywhere else except for Admin page?

Reply:
I’m still not convinced that the molecules/atoms/etc pattern is cleaner/better than the ‘feature’ pattern. Maybe for generic things, but it kind of feels like it takes more cognitive effort and ends up being somewhat arbitrary anyway.

In past projects I’ve stuck more to a ‘feature’ based folder structure, and then within that would sometimes break it up further by components/etc folders. We’ve vaguely used this pattern in a couple of places already for these more ‘private’ components.

—-

I definitely think we should be splitting up the admin stuff way more than it is (particularly the files that have tons of components in them), but I’d be inclined to think of it more in this ‘private component folder’ layout. It’d also be nice to have an explicit distinction between admin v1 and v2 in those folders. IMO that would logically lend itself more towards how we (or at least iI) tend to think about the app and it’s logical dependencies; and sort of aligns closer to the various routes/‘areas’ of the app.

Reply:
(This stuff is probably OOS to be discussing on this PR thread, and should probably be copied into a new backlog issue for further discussion/refinement/to assess our patterns/etc)

Reply:
Okay, I will refactor it for atoms for now and file a new issue to look into it at a later stage

Reply:
@TroySK This still seems to be outstanding?

Reply:
Yes, will make the change once https://github.com/sparkletown/sparkle/pull/1305#discussion_r617310471 gets done. Hoping to use that to generate the options.